### PR TITLE
Update lite build in documentation

### PR DESCRIPTION
**Pull request by martinRenou**
- Switch to using jupyterlite-xeus
- ~~Install dev version of jupyterlab-blocky in documentation~~ Does not work actually

### CODE REVIEW

**Comment by martinRenou** [comment]
This seems to be working now:

Screenshot from 2024-02-14 14-42-06

I see two issues though:
- the blockly UI is not in english?
- the kernel UI does not show up

Screenshot from 2024-02-14 15-02-48

**Comment by DenisaCG** [comment]
Thank you for working on this @martinRenou! 

I'm not sure what the issue with the language or the UI is, but it might be coming from the version of the `blockly` library we are using. I will look into it.

**Comment by martinRenou** [comment]
Thanks!

**Comment by DenisaCG** [comment]
>  the blockly UI is not in english?

Did you use the development install as`pip install -e . ".[dev]"`? 
And was your JupyterLab originally set in English? Did you have other language packages installed? @martinRenou

**Comment by martinRenou** [comment]
Sorry I did not clarify, the screenshot was taken from the readthedocs deployment for this PR:

https://jupyterlab-blockly--83.org.readthedocs.build/en/83

This build uses the environment https://github.com/QuantStack/jupyterlab-blockly/pull/83/files#diff-4c1832e89e371aa749c45d5942a4b294b068f05c43e1ffc2e16457331c3d020eR1

**Comment by DenisaCG** [comment]
After some debugging, the error with the `JupyterLite` deployment of `jupyterlab-blockly` seems to come from the version of `JupyterLab` it's using, as the function from the kernel UI seems to have changed in the later versions. In our extension we currently set up the kernel commands like this:

https://github.com/QuantStack/jupyterlab-blockly/blob/de7fbb8760f2118857c977a05a65c51243c3e383/packages/blockly-extension/src/index.ts#L217-L243

but in `JupyterLab` since `v4.0.0` they are set up like this:

https://github.com/jupyterlab/jupyterlab/blob/bb8a4edbe6fdfcd8abb763b414adf37e84aa2f08/packages/console-extension/src/index.ts#L836-L856

As a temporary fix, so the `JupyterLite` deployment works, we can pin the version of `JupyterLab` to be the latest compatible one, which would mean adding the following lines to `docs/build-enviroment.yml`:

```
- python=3.9
- jupyterlab=3.6.7
```

I have tried this locally and it should fix the kernel UI error.

Is there another way you recommend we deal with this for now?

In any case, I think migrating `jupyterlab-blockly` to `v4` of JupyterLab should be done soon and would be the best solution in the long run :)

File: docs/build-environment.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: jupyterlab-blockly-docs
+
+channels:
+  - conda-forge
+
+dependencies:
+  - mamba
+  - myst-parser
+  - pydata-sphinx-theme
```
Comment (DenisaCG) [inline docs/build-environment.yml:9]:
```suggestion
  - pydata-sphinx-theme
  - python=3.9
  - jupyterlab=3.6.7
```

Comment (martinRenou) [reply docs/build-environment.yml:9]: ~~Sounds good! Thanks for investigating!~~ Ah that cannot work unfortunately. This does not control the jupyterlab version jupyterlite is using

**Comment by martinRenou** [comment]
Thanks for investigating this!

Unfortunately using the latest jupyterlite we're stuck with jupyterlab v4, so we need to downgrade jupyterlite for now. Let's see if that works.

**Comment by martinRenou** [comment]
We need a release of jupyterlite-xeus with https://github.com/jupyterlite/xeus/pull/73 and I believe we should be good to go

**Comment by martinRenou** [comment]
It's working again! https://jupyterlab-blockly--83.org.readthedocs.build/en/83

Screenshot from 2024-02-20 10-14-21

**Comment by martinRenou** [comment]
@DenisaCG let's get this merged and we can fix the check-release workflow separately

**Comment by DenisaCG** [comment]
Great, thanks Martin!

**Pull request merged**